### ISSUE TITLE
Auto-scaling pods on the basis of messages in the queue with KEDA for processor intensive long running tasks..

### ISSUE THREAD

**Issue by vivek-sciffer** [kedacore/keda#3178]
### Discussed in https://github.com/kedacore/keda/discussions/3177

<div type='discussions-op-text'>

<sup>Originally posted by **vivek-sciffer** June 15, 2022</sup>
**My requirement is to have long running processor intensive tasks to be performed in parallel**. As working stack am using **(Kubernetes, KEDA, Azure Storage Queue, Dockerized Azure funtions/Python scripts).** To perform the task I intent, I used kubernetes **deployments** with **scaledObjects** and Kubernetes **jobs** with **scaledJobs**. 

I am providing yaml file for my work till now with the issues faced with both.

**ScaledObjects and Kubernetes Deployments**

```
data:
  FUNCTIONS_WORKER_RUNTIME: <---->
  AzureWebJobsStorage: <---->
  AzureDevStorage: <---->
apiVersion: v1
kind: Secret
metadata:
  name: thumbnail-processor
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: thumbnail-processor
  labels:
    app: thumbnail-processor
spec:
  selector:
    matchLabels:
      app: thumbnail-processor
  template:
    metadata:
      labels:
        app: thumbnail-processor
    spec:
      containers:
      - name: thumbnail-processor
        image: <---->
        env:
        - name: AzureFunctionsJobHost__functions__0
          value: thumbnail-processor
        envFrom:
        - secretRef:
            name: thumbnail-processor
---
apiVersion: keda.sh/v1alpha1
kind: ScaledObject
metadata:
  name: thumbnail-processor
  labels: {}
spec:
  scaleTargetRef:
    name: thumbnail-processor
  triggers:
  - type: azure-queue
    metadata:
      direction: in
      queueName: messages
      queueLength: "1"
      connectionFromEnv: AzureDevStorage
      endpointSuffix: <---->
---
```

This is scaling appropriately, but the pods created by KEDA are dying in 5-6 min on average, some pods live for 8-9 minutes too. But not more than that. This is resulting in repeated processing of messages and termination of long processes.

**ScaledJobs and Kubernetes Jobs**

```
apiVersion: v1
kind: Secret
metadata:
  name: queue-storage-connection-string
data:
  connection-string: <connection string>
  AzureDevStorage: <connection>
  proxy-job-container-pull: <container pull secret>
---
apiVersion: keda.sh/v1alpha1
kind: TriggerAuthentication
metadata:
  name: az-queue-auth
spec:
  secretTargetRef:
    - key: connection-string
      name: queue-storage-connection-string
      parameter: connection
---
apiVersion: keda.sh/v1alpha1
kind: ScaledJob
metadata:
  name: reflexion-proxy-job
  namespace: default
spec:
  jobTargetRef:
    parallelism: 1                            
    completions: 1                            
    activeDeadlineSeconds: 60000                
    backoffLimit: 3                           
    template:
      metadata:
        annotations:
          cluster-autoscaler.kubernetes.io/safe-to-evict: "false"
      spec:
        containers:
          - name: reflexion-proxy-job
            image: <container_image>
            imagePullPolicy: Always
            resources:
              limits:
                cpu: '500m'
              requests:
                cpu: '100m'
  maxReplicaCount: 100                  
  scalingStrategy:
    strategy: "custom"                         
    customScalingQueueLengthDeduction: 1      
    customScalingRunningJobPercentage: "0.5"  
    pendingPodConditions:                     
      - "Ready"
      - "PodScheduled"
      - "AnyOtherCustomPodCondition"
    multipleScalersCalculation : "max" 
  successfulJobsHistoryLimit: 1
  failedJobsHistoryLimit: 5
  pollingInterval: 2
  triggers:
    - type: azure-queue
      metadata:
        queueName: messages
        queueLenth: "1"
      authenticationRef:
        name: az-queue-auth
```

Now in Jobs, The Pods created by KEDA are  working fine. but scaling isn't up to the mark. The scaling is very lethargic. When bulk requests of 15-16 messages are put on the development environment. KEDA is creating 4 pods at start and then 2 jobs are running in parallel, where as my requirement is to processes all the requests in parallel as the message arrives in the queue. 

**Now, my question is. Is the technologies am using are appropriate to tackle and solve my issue, If yes then how can I solve my use case. and where am I making mistake.**
</div>

**Comment by tomkerkhove** [comment]
Closing as duplicate of https://github.com/kedacore/keda/discussions/3177
